src/main/java/org/folio/auth/authtokenmodule/storage/TokenStore.java

```diff
@@ -0,0 +1,60 @@
+package org.folio.auth.authtokenmodule.storage;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowSet;
+import io.vertx.sqlclient.SqlConnection;
+import io.vertx.sqlclient.Tuple;
+
+import org.folio.tlib.postgres.TenantPgPool;
+
+/**
+ * The base class for all token storage. Actual token store classes should
+ * extend this.
+ */
+public class TokenStore {
+  private static final Logger log = LogManager.getLogger(TokenStore.class);
+
+  protected Vertx vertx;
+  protected String tenant;
+  protected TenantPgPool pool;
+
+  public TokenStore(Vertx vertx, String tenant) {
+    this.vertx = vertx;
+    this.tenant = tenant;
+    this.pool = TenantPgPool.pool(vertx, tenant);
+  }
+
+
+  protected Future<Void> removeAll(String suffix) {
+    log.info("Removing all tokens from storage");
+
+    String delete = "DELETE FROM " + tableName(tenant, suffix);
+
+    return pool.preparedQuery(delete).execute().mapEmpty();
+  }
+
+  protected Future<Row> getRow(SqlConnection conn, String select, Tuple where) {
```

Comment:
The SQL query `SELECT is_redeemed, is_revoked FROM table WHERE id=$1` consists of
* the SELECT clause `SELECT is_redeemed, is_revoked`
* the FROM clause `FROM table`
* the WHERE clause `WHERE id=$1`

Tuple.of(id) is the parameter tuple.

Therefore I don't expect the method parameters `String select, Tuple where` but `String sql, Tuple tuple` or `String sql, Tuple parameters`.